### PR TITLE
improve the perf of matrix label population

### PR DESCRIPTION
**Pull request by AviAvni**
this fix improve perf of loading an rdb of graph created with the below query
GRAPH.QUERY x "UNWIND range(1, 1000000) AS x CREATE (:N1), (:N2), (:N3), (:N4), (:N5)"
master: 22.541 seconds
this fix: 3.536 seconds

### CODE REVIEW

**Review by jeffreylovitz — APPROVED** [review]
:+1: Great speedup!

File: src/serializers/graph_extensions.c [hunk 1]
```diff
@@ -82,9 +83,11 @@ void Serializer_Graph_SetNodeLabels
 
 		GxB_Vector_diag(v, m, 0, NULL);
 
-		GxB_Col_subassign(node_labels_m, NULL, NULL, v, GrB_ALL, 0, i, NULL);
+		GxB_Row_subassign(node_labels_m, NULL, NULL, v, i, GrB_ALL, 0, NULL);
 	}
 
+	GrB_transpose(node_labels_m, NULL, NULL, node_labels_m, NULL);
```
Comment (swilly22) [inline src/serializers/graph_extensions.c:89]: Perfect, we should have realized `GxB_Col_subassign` is expensive when `node_labels_m` is `CSR`

File: src/serializers/graph_extensions.c [hunk 2]
```diff
@@ -62,6 +62,7 @@ void Serializer_Graph_SetNode
 
 // computes NodeLabelMarix out of label matrices
 // NodeLabelMatrix[:i] = diag(LabelMatrix[i])
+// must be called once after all virtual keys loaded for perf
```
Comment (swilly22) [inline src/serializers/graph_extensions.c:65]: Try adding an assertion enforcing this restriction

**Review by swilly22 — COMMENTED** [review]
can we add a benchmark guarding from future possible performance degradation ?

File: src/serializers/graph_extensions.c (hunk 1 above)
Comment (filipecosta90) [reply src/serializers/graph_extensions.c:89]: > can we add a benchmark guarding from future possible performance degradation ?

image

@swilly22 we keep track of a "generic" rdb load timetaken over time that we can use to create the comparison tables but I believe we need a more advanced way of keeping track of this plus the setups with replication enabled ( that we don't have on performance automation still ). 
I will do the required changes on the performance package parts and then we can be more assertive in making sure te rdb load time and full sync time, etc... are kept at a constant level or improved

File: src/serializers/graph_extensions.c [hunk 3]
```diff
@@ -74,6 +75,12 @@ void Serializer_Graph_SetNodeLabels
 	RG_Matrix node_labels   = Graph_GetNodeLabelMatrix(g);
 	GrB_Matrix node_labels_m = RG_MATRIX_M(node_labels);
 
+#if RG_DEBUG
+	GrB_Index nvals;
+	GrB_Matrix_nvals(&nvals, node_labels_m);
+	ASSERT(nvals == 0);
+#endif
```
Comment (swilly22) [inline src/serializers/graph_extensions.c:82]: @AviAvni 
this is the right check to perform but located at the wrong place, it should be enforced by the caller,
can you please move it to: src/serializers/decoders/current/v11/decode_graph.c
right before calling `Serializer_Graph_SetNodeLabels(g);` ?

**Pull request merged**